src/app/clinic-dashboard/hiv/surge-report/surge-report.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { SurgeReportBaseComponent } from 'src/app/hiv-care-lib/surge-report/surge-report-base.component';
+import { SurgeResourceService } from 'src/app/etl-api/surge-resource.service';
+
+@Component({
+  selector: 'surge-report',
+  templateUrl: '../../../hiv-care-lib/surge-report/surge-report-base.component.html'
+})
+export class SurgeReportComponent extends SurgeReportBaseComponent implements OnInit {
+
+  public params: any;
+  public surgeReportSummaryData: any = [];
+  public columnDefs: any = [];
+
+  public statusError = false;
+  public errorMessage = '';
+  public showInfoMessage = false;
+  public isLoading = false;
+
+  constructor(public router: Router,
+    public route: ActivatedRoute,
+    public surgeReport: SurgeResourceService) {
+    super(router, route, surgeReport);
+  }
+
+  ngOnInit() {
+    this.route.queryParams.subscribe(
+      (params: any) => {
+        switch (params.currentView) {
+          case 'daily':
+            if (params) {
```

Comment:
it's more common for people to combine the 2 null checks under one if statement i.e if(params && params._date) { ... }